packages/now-python/now_init.py

```diff
@@ -40,7 +39,7 @@ def now_handler(event, context):
         _thread.start_new_thread(server.handle_request, ())
 
         payload = json.loads(event['body'])
-        path = unquote(payload['path'])
+        path = payload['path']
```

Comment:
Thanks for the PR!

It looks like this is causing a test to fail, `10-url-params-http-handler`, with the following error:

```
Fetched page https://test-3f5ng0zm5.now.sh/?hello=/ does not contain path=/?hello=/. Instead it contains path=/?hello=%2F
```

This was added in [PR 525](https://github.com/zeit/now-builders/pull/525) so perhaps we have to handle the query string differently than the path.

Reply:
If I remember it correctly, it's needed to handle encoding in query string properly. If it's causing `path` trouble, then I agree with @styfle, we will need to handle it differently (`query string` vs `path`).

cc @c0rvax

Reply:
## Transparency

I am no expert in HTTP, but from my perspective as a `now` user, I would say that the behaviour of my hosted code in `now` should be super similar to the behaviour of the exact same code launched with [http.server. HTTPServer](https://docs.python.org/3/library/http.server.html) locally

Next is what I did: 

1. Run this snippet https://gist.github.com/c0rvax/6f1fc3f170016a9c1350847c06ce2b47
2. Request `http://127.0.0.1:5001/?hello=/` from both: browser and curl
3. Check the terminal output which says `/?hello=/`

So for me it looks like the test is incorrect. As it in the first place changes the behaviour of the apps using BaseHTTPRequestHandler which no longer makes `now` transparent

## Simplicity of BaseHTTPRequestHandler

Another thought which i have come up with is that it looks like BaseHTTPRequestHandler is a very simple class which is not responsible for query encoding itself, neither it parses the parameters of the query. Users are on their own here. So i would say that there is no direct need in helping users with that. 

I might be wrong here, but reading some articles, I would say, that is the right way to do that

Reply:
@c0rvax I think there is a misunderstanding.

The test in question is specifically checking for `/?hello=/` which matches your example code snippet. This is currently working in the latest Now CLI.

However, this PR (3183) broke that test and instead returns `/?hello=%2F` as seen in Circle CI. That makes sense that it would break because [PR 525](https://github.com/zeit/now-builders/pull/525) added `unquote` to fix this behavior and thus removing `unquote` will break it.

The name of `payload['path']` is a bit misleading because it appears to be more than just the `pathname`. It also includes the `search` string and `hash`. (See this URL breakdown 
http://bl.ocks.org/abernier/3070589)

### Solution

So I believe the correct solution is to keep the `pathname` encoded and only decode (unquote) the `search` string and `hash`. Does that make sense?

Reply:
Okay, i got it now! I am trying to understand the origin of the `/` being encoded on the now side. `fetch` which is used in tests does not do any encoding to that: i have checked that. Looking deeper

Reply:
The `event` object is not capable of being changed. The originates from the routing layer so every builder receives the same `event` object and its up to the builder to turn in into a pretty API for the user.

The tests are using `fetch` so simulate the requests from the browser.

I would suggest starting with adding a new test fixture for your particular use case (URLs with spaces) and then update the builder to handle this scenario (probably parse the `path` into a `pathname` and a `search` string).